src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -6031,8 +6031,10 @@ void MergeTreeData::reportBrokenPart(MergeTreeData::DataPartPtr & data_part) con
                 broken_part_callback(part->name);
         }
     }
-    else
+    else if (data_part && data_part->getState() == IMergeTreeDataPart::State::Active)
```

Comment:
Maybe we should also check `State::PreActive`? For example, we fetched/merged a part, but failed to commit it because of different checksums. It should be in `PreActive` state and we should check it?

Reply:
I'm not sure, such part will be removed if we failed to commit it, so there will be nothing to check.